test/integration_test/clusterdomain_test.go

```diff
@@ -146,6 +146,10 @@ func testClusterDomainsFailover(
 	// Reduce the replicas on cluster 1
 	scaleFactor, err := schedulerDriver.GetScaleFactorMap(ctxs[0])
 	require.NoError(t, err, "Unexpected error on GetScaleFactorMap")
+
+	// Copy the old scale factor map
+	oldScaleFactor := scaleFactor
```

Comment:
This won't make a copy of the map, so oldScaleFactor will also have 0s for the scale after the for loop below

Reply:
Fixed.